client/my-sites/upgrades/cart/cart-item.jsx

```diff
@@ -1,29 +1,39 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+var React = require( 'react' ),
+	range = require( 'lodash/utility/range' );
 
 /**
  * Internal dependencies
  */
-var analytics = require( 'analytics' ),
+var analyticsMixin = require( 'lib/mixins/analytics' ),
 	canRemoveFromCart = require( 'lib/cart-values' ).canRemoveFromCart,
+	config = require( 'config' ),
 	cartItems = require( 'lib/cart-values' ).cartItems,
 	getIncludedDomain = cartItems.getIncludedDomain,
 	isCredits = require( 'lib/products-values' ).isCredits,
 	isDomainProduct = require( 'lib/products-values' ).isDomainProduct,
 	isGoogleApps = require( 'lib/products-values' ).isGoogleApps,
-	upgradesActions = require( 'lib/upgrades/actions' );
+	upgradesActions = require( 'lib/upgrades/actions' ),
+	isDomainRegistration = require( 'lib/products-values' ).isDomainRegistration;
 
 module.exports = React.createClass( {
 	displayName: 'CartItem',
 
+	mixins: [ analyticsMixin( 'cartItem' ) ],
+
 	removeFromCart: function( event ) {
 		event.preventDefault();
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Remove From Cart Icon', 'Product ID', this.props.cartItem.product_id );
+		this.recordEvent( 'remove', this.props.cartItem.product_id );
```

Comment:
I understand where you're coming from, but that's the pattern we've been using so far in everywhere, I don't think this PR is the right place to solve it.